### PR TITLE
Log errors to a Discord Channel

### PR DESCRIPTION
**Pull request by hemant2132**
Since the bot is hosted on repl.it, many-a-times it is not easy to get notified when an error occurs. Hence, logging the errors to a Discord channel will be helpful.

### CODE REVIEW

File: commands/events/addEvent.js [hunk 1]
```diff
@@ -86,16 +86,9 @@ module.exports = {
...
-          .then((doc) => {
-            str = "Event Successfully Added! 😀";
-          })
-          .catch((err) => {
-            str = "There was some error. 🙁";
-            console.log(err);
-          });
+        await event.save().then((doc) => {
+          str = "Event Successfully Added! 😀";
+        });
```
Comment (utkarsh261) [inline commands/events/addEvent.js]: No catching errors here?
If some error occurs, bot will try sending empty string as message to the channel.

Comment (hemant2132) [reply commands/events/addEvent.js]: If some error occurs then the _try_ execution will stop then and there and the control will flow to the _catch_ [here](https://github.com/Knuth-Programming-Hub/KLE-bot/blob/discord-server/bot.js#L103).
However, I have made a small change here in this [commit](https://github.com/Knuth-Programming-Hub/KLE-bot/pull/23/commits/7476b95232b40ebb21016f3800c16c852ad398bc). Please review this now. 😅

**Comment by utkarsh261** [comment]
LGTM. Major refactoring also got done by removing all the try catch blocks.

I'll merge this.

*But* does handling errors at the top level gives correct location of where a particular error occured at the first place?

**Comment by hemant2132** [comment]
> _But_ does handling errors at the top level gives correct location of where a particular error occured at the first place?

Yes. During testing one can simply use `console.log(error)` instead of `channel.send(error.toString())` for it. If one wants it in the Discord channel,  `channel.send(error.stack)` (refer [this](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/Stack)) can be used.

**Comment by utkarsh261** [comment]
> > _But_ does handling errors at the top level gives correct location of where a particular error occured at the first place?
> 
> Yes. During testing one can simply use `console.log(error)` instead of `channel.send(error.toString())` for it. If one wants it in the Discord channel,  `channel.send(error.stack)` (refer [this](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/Stack)) can be used.

Nice.
Shouldn't we always send the stack trace in logging channel?
Will provide more context and clarity in my opinion.

**Comment by hemant2132** [comment]
Oh yes, I also feel that. 
I will make this change in my next PR.  😉

**Pull request merged**